adapter/internal/ga/client.go

```diff
@@ -64,13 +64,14 @@ const (
 // APIEvent represents the event corresponding to a single API Deploy or Remove event
 // based on XDS state changes
 type APIEvent struct {
-	APIUUID       string
-	RevisionUUID  string
-	IsDeployEvent bool
+	APIUUID          string
+	RevisionUUID     string
+	IsDeployEvent    bool
+	OrganizationUUID string
```

Comment:
These are inside the internal package, internal package is only used by LA, so what would be the org ID when used on premise

Reply:
Yes. Because this event is only used by local adapter. In JMS scenario we read amqp delivery object, convert into an internal object proceed. Just like that from the XDS content, this event would be populated and pushed into a channel.

Reply:
This event is used by https://github.com/wso2/product-microgateway/pull/2127

Reply:
My concerns is we are bringing Choreo specific logic to the LA

Reply:
Sorry my answer was not complete. The orgID used in on premise scenario would be,

- In api deployment flow, it is fetched from the api.json (Hence we do not need for GA-LA flow as well, but added to maintain consistency)
- In api Remove from gateway flow, this org-id is picked from `tenant-domain` in the JMS event. (This is what we are trying to get by introducing this Org-id)
https://github.com/wso2/product-microgateway/blob/a0e2d70161b345935743ea7bbac311aae296f974/adapter/internal/messaging/notification_listener.go#L172